tests/php/LeftAndMainTest.php

```diff
@@ -180,4 +181,38 @@ public function testGetHelpLinks()
         $this->assertEquals('SilverStripe', $silverstripeLink['Title']);
         $this->assertEquals('www.silverstripe.org', $silverstripeLink['URL']);
     }
+
+    /**
+     * @dataProvider provideTestCMSVersionNumber
+     */
+    public function testCMSVersionNumber($frameworkVersion, $expected)
+    {
+        $versionProvider = $this
+            ->getMockBuilder(VersionProvider::class)
+            ->setMethods(['getModules', 'getModuleVersionFromComposer'])
+            ->getMock();
+        $data = ['silverstripe/framework' => $frameworkVersion];
+        $versionProvider->method('getModules')->willReturn($data);
+        $versionProvider->method('getModuleVersionFromComposer')->willReturn($data);
+        $leftAndMain = $this
+            ->getMockBuilder(LeftAndMain::class)
+            ->setMethods(['getVersionProvider'])
+            ->getMock();
+        $leftAndMain->method('getVersionProvider')->willReturn($versionProvider);
+        $this->assertSame($expected, $leftAndMain->CMSVersionNumber());
+    }
+
+    /**
+     * @return array
+     */
+    public function provideTestCMSVersionNumber()
+    {
+        return [
+            ['4.9.1', '4.9'],
+            ['4.10.5', '4.10'],
+            ['4.9.x-dev', '4.9'],
+            ['4.10.x-dev', '4.10'],
+            ['myfork', 'myfork'],
+        ];
```

Comment:
Can we test the 3-digit version as well? Thanks

```suggestion
        return [
            ['4.9.1', '4.9'],
            ['4.10.5', '4.10'],
            ['4.236.7, '4.236'],
            ['4.9.x-dev', '4.9'],
            ['4.10.x-dev', '4.10'],
            ['myfork', 'myfork'],
        ];
```

Reply:
Updated